nutils/evaluable.py

```diff
@@ -5029,6 +5029,7 @@ def eval_sparse(funcs: AsEvaluableArray, **arguments: typing.Mapping[str, numpy.
         yield data
 
 
+@functools.lru_cache(256)
```

Comment:
Have you considered a weakref-like module level dictionary, with keys formed of `tuple(ref(func, removeme) for func in funcs)` plus whatever else is relevant, and `removeme` a callback that removes the entry? That way it is triggered as soon as any involved function is garbage collected.

Reply:
The `lru_cache` is a temporary solution until there's a solver 2.0 that avoids repeated calls to `compile`. I've reduced the size of the cache to 32.